core/src/apps/management/apply_settings.py

```diff
@@ -167,17 +167,17 @@ async def _require_confirm_change_passphrase_source(
     await confirm_change_passphrase_source(passphrase_always_on_device)
 
 
-async def _require_confirm_change_display_rotation(rotation: int) -> None:
-    if rotation == 0:
+async def _require_confirm_change_display_rotation(rotation: DisplayRotation) -> None:
+    if rotation == DisplayRotation.North:
         label = TR.rotation__north
-    elif rotation == 90:
+    elif rotation == DisplayRotation.East:
         label = TR.rotation__east
-    elif rotation == 180:
+    elif rotation == DisplayRotation.South:
         label = TR.rotation__south
-    elif rotation == 270:
+    elif rotation == DisplayRotation.West:
         label = TR.rotation__west
     else:
-        raise DataError("Unsupported display rotation")
+        raise ValueError(f"Unsupported display rotation: {rotation}")
```

Comment:
use a `RuntimeError` here, this cannot happen because your if/elif is exhaustive

also don't use generic errors with texts https://docs.trezor.io/trezor-firmware/core/misc/exceptions.html

dtto below

Reply:
https://github.com/trezor/trezor-firmware/pull/4125/commits/677d33afba60e049a84e90778b7ce6df69fcfa7b